examples/quickstart/jupyter-notebooks/druidapi/tasks.py

```diff
@@ -0,0 +1,178 @@
+REQ_GET_TASK = REQ_POST_TASK + '/{}'
+REQ_TASK_STATUS = REQ_GET_TASK + '/status'
+REQ_TASK_REPORTS = REQ_GET_TASK + '/reports'
+REQ_END_TASK = REQ_GET_TASK
+REQ_END_DS_TASKS = REQ_END_TASK + '/shutdownAllTasks'
+
+class TaskClient:
+    """
+    Client for task-related APIs. The APIs connect through the Router to
+    the Overlord.
+    """
+    
+    def __init__(self, rest_client):
+        self.client = rest_client
+
+    def tasks(self, state=None, table=None, type=None, max=None, created_time_interval=None):
+        '''
+        Retrieve list of tasks.
+
+        Parameters
+        ----------
+        state : str, default = None
+        	Filter list of tasks by task state. Valid options are "running", 
+            "complete", "waiting", and "pending". Constants are defined for
+            each of these in the `consts` file.
+        table : str, default = None
+        	Return tasks filtered by Druid table (datasource).
+        created_time_interval : str, Default = None
+        	Return tasks created within the specified interval.
+        max	: int, default = None
+            Maximum number of "complete" tasks to return. Only applies when state is set to "complete".
+        type : str, default = None
+        	filter tasks by task type.
+
+        Reference
+        ---------
+        `GET /druid/indexer/v1/tasks`
+
+        See https://druid.apache.org/docs/latest/operations/api-reference.html#get-15
+        '''
+        params = {}
+        if state is not None:
+            params['state'] = state
+        if table is not None:
+            params['datasource'] = table
+        if type is not None:
+            params['type'] = type
+        if max is not None:
+            params['max'] = max
+        if created_time_interval is not None:
+            params['createdTimeInterval'] = created_time_interval
+        return self.client.get_json(REQ_TASKS, params=params)
+
+    def task(self, task_id):
+        """
+        Retrieve the "payload" of a task.
+
+        Parameters
+        ----------
+        task_id : str
+            The id of the task to retrieve
+
+        Reference
+        ---------
+        `GET /druid/indexer/v1/task/{taskId}`
+
+        See https://druid.apache.org/docs/latest/operations/api-reference.html#get-15
+        """
+        return self.client.get_json(REQ_GET_TASK, args=[task_id])
+
+    def task_status(self, task_id):
+        '''
+        Retrieve the status of a task.
+
+        Parameters
+        ----------
+        task_id : str
+            The id of the task to retrieve
+
+        Reference
+        ---------
+        `GET /druid/indexer/v1/task/{taskId}/status`
+
+        See https://druid.apache.org/docs/latest/operations/api-reference.html#get-15
+        '''
+        return self.client.get_json(REQ_TASK_STATUS, args=[task_id])
+
+    def task_reports(self, task_id):
+        '''
+        Retrieve a task completion report for a task.
+        Only works for completed tasks.
```

Comment:
Could be helpful to say what happens when called on incomplete tasks. "Incomplete tasks raise an exception"? Or return null?